corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -429,6 +429,20 @@ def get_reverse_indexed_cases(domain, case_ids):
         return cases
 
     @staticmethod
+    def get_extension_chain(domain, case_ids):
+        assert isinstance(case_ids, list)
+
+        extension_chain_ids = set()
+        incoming_extensions = set(CaseAccessorSQL.get_extension_case_ids(domain, case_ids))
+        all_extension_ids = set(incoming_extensions)
+        new_extensions = set(incoming_extensions)
+        while new_extensions:
+            new_extensions = set(CaseAccessorSQL.get_extension_case_ids(domain, list(new_extensions)))
+            all_extension_ids = all_extension_ids | new_extensions
```

Comment:
What's the difference between `all_extension_ids` and `extension_chain_ids`? I don't see the need for both.